modules/conflict_resolver/php/unresolved.class.inc

```diff
@@ -0,0 +1,333 @@
+    protected function supportedVersions() : array
+    {
+        return array('unknown');
+    }
+
+    /**
+     * Provide the select inputs options
+     *
+     * @return array Dynamic field options
+     */
+    private function _getFieldOptions() : array
+    {
+        $sites       = array_values(\Utility::getSiteList(false));
+        $visitlabels = array_values(\Utility::getVisitList());
+        $projects    = array_values(\Utility::getProjectList());
+
+        return array(
+                'site'       => array_combine($sites, $sites),
+                'instrument' => \Utility::getAllInstruments(),
+                'visitLabel' => array_combine($visitlabels, $visitlabels),
+                'project'    => array_combine($projects, $projects),
+               );
+    }
+
+    /**
+     * Handles a request to Resolved conflicts
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+
+        case 'POST':
+            return $this->_handlePOST($request);
+
+        case 'OPTIONS':
+            return (new \LORIS\Http\Response())
+                ->withHeader('Allow', $this->allowedMethods());
+
+        default:
+            return new \LORIS\Http\Response\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+        }
+    }
+
+    /**
+     * Create an array representation of this endpoint's response body
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface
+     */
+    private function _handleGET(ServerRequestInterface $request): ResponseInterface
+    {
+        if (!isset($this->_cache)) {
+            $user        = $request->getAttribute('user');
+            $provisioner = (new UnresolvedProvisioner())->forUser($user);
+
+            $conflicts = (new \LORIS\Data\Table())
+            ->withDataFrom($provisioner)
+            ->toArray($user);
+
+            $body = array(
+                     'data'         => $conflicts,
+                     'fieldOptions' => $this->_getFieldOptions(),
+                    );
+
+            $this->_cache = new \LORIS\Http\Response\JsonResponse($body);
+        }
+        return $this->_cache;
+    }
+
+    /**
+     * Handle POST request to fix conflicts. It will instantiate the instruments
+     * and change the filed name to the correct value, then rescore them. It will
+     * also remove the conflict_unresolved row and add a conflicts_resolved one.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface
+     */
+    private function _handlePOST(ServerRequestInterface $request): ResponseInterface
+    {
+        $user = $request->getAttribute('user');
```

Comment:
```suggestion
        $user = \NDB_Factory::singleton()->user();
```
I'm not familiar with this module but maybe the user could be grabbed this way instead? This could have some security issues if someone alters the user in the request.

Let me know if I'm missing something.

Reply:
The user is added to the request in `BaseRouter.php`, and I prefer having the value passed by dependency injection instead of making a static call to NDB_Factory. using the one from the request makes it easier to test with different users for instance.